scipy/stats/stats.py

```diff
@@ -4275,6 +4277,158 @@ def mannwhitneyu(x, y, use_continuity=True):
     return MannwhitneyuResult(smallu, distributions.norm.sf(z))
 
 
+# TODO: add paired=False, do a signed-rank test if paired=True or y is not provided. See morestats.wilcoxon
+def mww(x, y, correction=True, exact='auto',
+             alternative='two-sided'):
+    """
+    Computes two-sample unpaired Mann-Whitney-Wilcoxon tests.
+
+    Parameters
+    ----------
+    x : array_like
+        The first set of measurements.
+    y : array_like
+        The second set of measurements.
+    correction : bool, optional
+        If True, apply continuity correction by adjusting the Wilcoxon rank
+        statistic by 0.5 towards the mean value when computing the z-statistic.
+        Default is False.
+    exact : bool, optional
+        Whether an exact test should be performed on the data. See notes for
+        default behavior.
+    alternative : string, optional
+        Whether a two-tailed or one-tailed test should be performed on the
+        supplied vectors. Arguments are 'two-tailed', 'less', and 'greater'.
+        Default is 'two-tailed'.
+
+    Returns
+    -------
+    Bunch object with following attributes:
+        statistic : float
+            The test statistic.
+        pvalue : float
+            The pvalue of the test.
+        exact : bool
+            Indicates if an exact pvalue was calculated.
+        alternative : string
+            Describes the alternative hypothesis.
+        u1 : float
+            The U-value corresponding to the set of measurements in x
+        u2 : float
+            The U-value corresponding to the set of measurements in y
+
+    Notes
+    -----
+    Exact tests should be used for smaller sample sizes. Concretely, as
+    len(x) and len(y) increase to and beyond 8, the distribution of U differs
+    negligibly from the normal distribution[1]. The default behavior of this
+    test is to calculate the number of possible sequences for inputs of
+    length(x) and length(y), and to do an exact calculation if the number of
+    possible combinations is <100000. The default behavior may be overridden
+    with the use_exact flag.
+
+    This test corrects for ties and by default uses a continuity correction
+    when approximating the U statistic distribution.
+
+    The reported p-value is for a two-sided hypothesis. To get the one-sided
+    p-value set alternative to 'greater' or 'less' (default is 'two-sided').
+
+    For Mann-Whitney-Wilcoxon tests, the reported U statistic is the U used to
+    test the hypothesis. The u1 and u2 statistics are returned as well,
+    corresponding to x and y, respectively.
+
+
+    .. versionadded:: 0.17.0
+
+    References
+    ----------
+    .. [1] HB Mann and DR Whitney (1947). "On a Test of Whether one of Two
+            Random Variables is Stochastically Larger than the Other".
+            Annals of Mathematical Statistics. doi:10.1214/aoms/1177730491
```

Comment:
Can you format this like so:

    H.B. Mann and D.R. Whitney, "On a test of whether one of two random variables is
    stochastically larger than the other", The Annals of Mathematical Statistics, 
    Vol. 18, pp. 50-60, 1947.

Note that this is the standard formatting for all citations (see https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt)